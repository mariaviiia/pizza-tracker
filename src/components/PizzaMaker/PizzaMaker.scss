.pizza-maker {
  &__container {
    display: flex;
    height: 200px;
  }
  &__title {
    font-size: 15px;
    font-family: cursive;
    font-weight: bold;
    letter-spacing: 0.5px;
    text-align: center;
    padding: 1em;
    background-color: rgba(175, 103, 45, 0.4);
    text-shadow: 1px 1px #fff;
  }
  &__pizza-name {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 1em;
    span {
      font-family: cursive;
      font-size: 12px;
      font-weight: bold;
      letter-spacing: 0.5px;
    }
    input {
      width: 200px;
      margin-top: 10px;
      border: 1px solid gray;
      border-radius: 2px;
      &:focus {
        outline: none;
        box-shadow: 1px 3px 3px rgba(180, 0, 15, 0.1);
      }
    }
  }
  &__buttons {
    padding: 1em;
    margin-top: 1em;
    width: auto;
    display: flex;
    justify-content: flex-end;
    &--pay {
      margin-left: 15px;
    }
  }
}
