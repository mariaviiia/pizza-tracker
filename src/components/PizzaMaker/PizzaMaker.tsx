import { Link } from "react-router-dom";
import Button, { BUTTON_TYPES } from "../common/Button/Button";
import IngredientList from "../common/IngredientList/IngredientList";
import Pizza from "../common/Pizza/Pizza";
import { createContext, useContext, useState } from "react";
import { IngredientModel } from "../common/Ingredient/Ingredient";
import Prices from "../common/Prices/Prices";
import { PizzasContext } from "../../Routes/Routes";
import "./PizzaMaker.scss";

export const SelectedIngredientsContext = createContext<{
  ingredients: IngredientModel[];
  setIngredients: (newIngredients: IngredientModel[]) => void;
  selectedIngredients: IngredientModel[];
  setSelectedIngredients: (newSelectedIngredients: IngredientModel[]) => void;
}>([] as any);

const PizzaMaker = () => {
  const [selectedIngredients, setSelectedIngredients] = useState<
      IngredientModel[]
    >([]),
    [ingredients, setIngredients] = useState<IngredientModel[]>([]),
    { pizzas, setPizzas } = useContext(PizzasContext),
    [pizzaName, setPizzaName] = useState<string>(""),
    addPizza = () => {
      setPizzas([
        ...pizzas,
        {
          name: pizzaName,
          ingredients: [...selectedIngredients],
        },
      ]);
      setPizzaName("");
      setSelectedIngredients([]);
      setIngredients(
        ingredients.map((ingredient) => ({
          ...ingredient,
          isSelected: false,
        }))
      );
    };

  return (
    <SelectedIngredientsContext.Provider
      value={{
        selectedIngredients,
        setSelectedIngredients,
        ingredients,
        setIngredients,
      }}
    >
      <div className="pizza-maker">
        <div className="pizza-maker__title">Make your pizza</div>
        <div className="pizza-maker__container">
          <Pizza />
          <IngredientList />
        </div>
        <Prices />
        <div className="pizza-maker__pizza-name">
          <span>Which is your favorite city?</span>
          <input
            value={pizzaName}
            onChange={({ target }) => setPizzaName(target.value)}
          />
        </div>
        <div className="pizza-maker__buttons">
          <Button
            type={BUTTON_TYPES.SECONDARY}
            disabled={!pizzaName}
            buttonLabel="Add another pizza"
            onClick={addPizza}
          />
          <div className="pizza-maker__buttons--pay">
            <Link to="/userInfo">
              <Button
                onClick={addPizza}
                buttonLabel="PAY"
                disabled={!pizzaName}
              />
            </Link>
          </div>
        </div>
      </div>
    </SelectedIngredientsContext.Provider>
  );
};

export default PizzaMaker;
