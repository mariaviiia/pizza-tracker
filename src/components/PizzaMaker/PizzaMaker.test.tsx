import PizzaMaker, { SelectedIngredientsContext } from "./PizzaMaker";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

describe("PizzaMaker component", () => {
  const MockSelectedIngredients = {
    selectedIngredients: [],
    setSelectedIngredients: () => undefined,
    ingredients: [],
    setIngredients: () => undefined,
  };
  it("renders correctly", () => {
    const { container } = render(
      <SelectedIngredientsContext.Provider value={MockSelectedIngredients}>
        <Router>
          <PizzaMaker />
        </Router>
      </SelectedIngredientsContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
