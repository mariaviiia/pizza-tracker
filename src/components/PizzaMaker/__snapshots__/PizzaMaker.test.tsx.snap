// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PizzaMaker component renders correctly 1`] = `
<div>
  <div
    class="pizza-maker"
  >
    <div
      class="pizza-maker__title"
    >
      Make your pizza
    </div>
    <div
      class="pizza-maker__container"
    >
      <ul
        class="pizza__container"
      >
        <span
          class="pizza__title"
        >
          Selected Ingredients
        </span>
        <li>
          <div>
            <div
              class="pizza"
            >
              <img
                alt=""
                class="pizza__icon"
                src="dough.svg"
              />
              <span
                class="pizza__name"
              >
                Dough
              </span>
            </div>
          </div>
        </li>
      </ul>
      <div
        class="ingrendient-list"
      >
        <ul>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/cheese.svg"
                />
                <span
                  class="ingredient__name"
                >
                  cheese
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/pepperoni.svg"
                />
                <span
                  class="ingredient__name"
                >
                  pepperoni
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              4500
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/pineapple.svg"
                />
                <span
                  class="ingredient__name"
                >
                  pineapple
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/mushroom.svg"
                />
                <span
                  class="ingredient__name"
                >
                  mushroom
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/ham.svg"
                />
                <span
                  class="ingredient__name"
                >
                  ham
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/sausage.svg"
                />
                <span
                  class="ingredient__name"
                >
                  sausage
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/pepper.svg"
                />
                <span
                  class="ingredient__name"
                >
                  pepper
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/tuna.svg"
                />
                <span
                  class="ingredient__name"
                >
                  tuna
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/olive.svg"
                />
                <span
                  class="ingredient__name"
                >
                  olive
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/chicken.svg"
                />
                <span
                  class="ingredient__name"
                >
                  chicken
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              5000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/bacon.svg"
                />
                <span
                  class="ingredient__name"
                >
                  bacon
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/tomato.svg"
                />
                <span
                  class="ingredient__name"
                >
                  tomato
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/shrimp.svg"
                />
                <span
                  class="ingredient__name"
                >
                  shrimp
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/meat.svg"
                />
                <span
                  class="ingredient__name"
                >
                  meat
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
          <li
            class="ingredient__item ingredient"
          >
            <input
              type="checkbox"
            />
            <div
              class="ingredient__container undefined"
            >
              <div
                class="ingredient"
              >
                <img
                  alt=""
                  class="ingredient__icon"
                  src="/assets/ingredients/salami.svg"
                />
                <span
                  class="ingredient__name"
                >
                  salami
                </span>
              </div>
            </div>
            <span
              class="ingredient__price"
            >
              $ 
              3000
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="prices"
    >
      <ul>
        <li>
          <span>
            base:
          </span>
           10000
        </li>
        <li
          class="total"
        >
          <span>
            total:
          </span>
          10000
        </li>
      </ul>
    </div>
    <div
      class="pizza-maker__pizza-name"
    >
      <span>
        Which is your favorite city?
      </span>
      <input
        value=""
      />
    </div>
    <div
      class="pizza-maker__buttons"
    >
      <button
        class="button button-secondary button__size-md"
        disabled=""
      >
        Add another pizza
      </button>
      <div
        class="pizza-maker__buttons--pay"
      >
        <a
          href="/userInfo"
        >
          <button
            class="button button-primary button__size-md"
            disabled=""
          >
            PAY
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
